File: Orbit.Domain/DTOs/Finance/HistoricData/GetHistoricMonthDataDto.cs
namespace Orbit.Domain.DTOs.Finance.HistoricData
{
    public class GetHistoricMonthDataDto
    {
        public required decimal TotalSpent { get; set; }
        public required decimal TotalSaved { get; set; }
        public required decimal AmountLeftOver { get; set; }
        public required MonthlySpendingDataBreakdown SpendingDataBreakdown { get; set; }
        public required TopSpendingMerchantsData[] TopSpendingMerchants { get; set; }
        public required TopTransactionsData[] TopTransactions { get; set; }
        public required SpendingPerDayData[] SpendingPerDay { get; set; }
    }

    public class MonthlySpendingDataBreakdown
    {
        public required PotSpendingData[] PotSpendings { get; set; }
    }

    public class TopSpendingMerchantsData
    {
        public required string MerchantName { get; set; }
        public required int TransactionsCount { get; set; }
        public required decimal AmountSpent { get; set; }
    }

    public class TopTransactionsData
    {
        public required string MerchantName { get; set; }
        public required string TransactionDate { get; set; }
        public required decimal AmountSpent { get; set; }
        public string? PotName { get; set; }
    }

    public class SpendingPerDayData
    {
        public required string Date { get; set; }
        public required decimal AmountSpent { get; set; }
    }

    public class PotSpendingData
    {
        public required string PotName { get; set; }
        public required decimal AmountSpent { get; set; }
        public required decimal PercentageOfTotalSpent { get; set; }
    }
}

File: orbit.web/interfaces/api/historicData/GetHistoricMonthDataDto.ts
export interface GetHistoricMonthDataDto {
  totalSpent: number
  totalSaved: number
  amountLeftOver: number
  spendingDataBreakdown: MonthlySpendingDataBreakdown
  topSpendingMerchants: TopSpendingMerchantsData[]
  topTransactions: TopTransactionsData[]
  spendingPerDay: SpendingPerDayData[]
}

export interface MonthlySpendingDataBreakdown {
  potSpendings: PotSpendingData[]
}

export interface TopSpendingMerchantsData {
  merchantName: string
  transactionsCount: number
  amountSpent: number
}

export interface TopTransactionsData {
  merchantName: string
  transactionDate: string
  amountSpent: number
  potName?: string
}

export interface SpendingPerDayData {
  date: string
  amountSpent: number
}

export interface PotSpendingData {
  potName: string
  amountSpent: number
  percentageOfTotalSpent: number
}
